File: File Templates/VIPER/Viper View Controller.xctemplate/Objective-C/___FILEBASENAME___Router.m
//___FILEHEADER___
 
#import "___VARIABLE_ModuleName___Router.h"
#import "___VARIABLE_ModuleName___VC.h"
#import "___VARIABLE_ModuleName___Interactor.h"
#import "___VARIABLE_ModuleName___Presenter.h"
 
@implementation ___VARIABLE_ModuleName___Router

+ (UIViewController *)createModule
{
    
    /**
     * Berfungsi untuk menghubungkan komponen VIPER per Modul
     */
    
    ___VARIABLE_ModuleName___VC *viewController = [self.mainStoryboard instantiateViewControllerWithIdentifier:@"___VARIABLE_ModuleName___VC"];
    ___VARIABLE_ModuleName___Interactor *interactor = [[___VARIABLE_ModuleName___Interactor alloc] init];
    ___VARIABLE_ModuleName___Router *router = [[___VARIABLE_ModuleName___Router alloc] init];
    ___VARIABLE_ModuleName___Presenter *presenter = [[___VARIABLE_ModuleName___Presenter alloc] initWithInterface:viewController interactor:interactor router:router];
    
    /// Menghubungkan
    viewController.presenter = presenter;
    router.viewController = viewController;
    
    return viewController;
}

+ (UIStoryboard *)mainStoryboard
{
    
    /**
     * Berfungsi untuk mengembalikan nilai storyboard
     * Untuk nama storyboard, dikondisikan
     */
    
    return [UIStoryboard storyboardWithName:@"Main" bundle:NSBundle.mainBundle];
}
 
@end

File: File Templates/Viper View Controller.xctemplate/Swift/___FILEBASENAME___Router.swift
//___FILEHEADER___

import UIKit

final class ___VARIABLE_ModuleName___Router: ___VARIABLE_ModuleName___RouteProtocol {

    // MARK: - Create Modul
    
    class func createModule() -> UIViewController {
        let view = mainstoryboard.instantiateViewController(withIdentifier: "___VARIABLE_ModuleName___VC") as! ___VARIABLE_ModuleName___VC
        let presenter: ___VARIABLE_ModuleName___PresenterProtocol & ___VARIABLE_ModuleName___InteractorOutputProtocol = ___VARIABLE_ModuleName___Presenter()
        let interactor: ___VARIABLE_ModuleName___InteractorInputProtocol = ___VARIABLE_ModuleName___Interactor()
        let router: ___VARIABLE_ModuleName___RouteProtocol = ___VARIABLE_ModuleName___Router()

        // Conecting
        view.presenter = presenter
        presenter.view = view
        presenter.wireFrame = router
        presenter.interactor = interactor
        interactor.presenter = presenter

        return view
    }

    // MARK: - Static variable for storyboard

    static var mainstoryboard: UIStoryboard {
        return UIStoryboard(name:"Main", bundle: Bundle.main)
    }
}
